src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,970 @@
+                _filter_out_nones(encoder_out.encoder_out, encoder_out.encoder_states, encoder_out.encoder_attn),
+            )
+        if self.output_hidden_states and self.output_attentions:
+            return (
+                dec_features,
+                dec_hidden,
+                dec_attn,
+                encoder_out.encoder_out,
+                encoder_out.encoder_states,
+                encoder_out.encoder_attn,
+            )
+        elif self.output_hidden_states:
+            return (
+                dec_features,
+                dec_hidden,
+                encoder_out.encoder_out,
+                encoder_out.encoder_states,
+            )
+        elif self.output_attentions:
+            return (dec_features, dec_attn, encoder_out.encoder_out, encoder_out.encoder_attn)
+        else:
+            return (dec_features, encoder_out.encoder_out)
+
+    @staticmethod
+    def shift_tokens_left(input_ids, pad_token_id):
+        """Shift input ids one token to the left"""
+        prev_output_tokens = input_ids.clone()
+        prev_output_tokens[:, 0] = input_ids.gather(
+            1, (input_ids.ne(pad_token_id).sum(dim=1) - 1).unsqueeze(-1),
+        ).squeeze()
+        prev_output_tokens[:, 1:] = input_ids[:, :-1]
+        return prev_output_tokens
+
+    def max_positions(self):
+        """Maximum length supported by the model."""
+        return (self.encoder.max_positions(), self.decoder.max_positions())
+
+
+class BartForMaskedLM(PretrainedBartModel):
+    base_model_prefix = "model"
+
+    def __init__(self, config: BartConfig):
+        super().__init__(config)
+        self.model = BartModel(config)
+
+    def forward(self, *args, **kwargs):  # TODO(SS): better signature when base API decided
+        kwargs["return_for_head"] = True
+        lm_labels = kwargs.pop("lm_labels", None)
+        decoder_outputs, encoder_outputs = self.model(*args, **kwargs)
+        lm_logits = self.output_layer(decoder_outputs[0])
+        decoder_outputs = (lm_logits,) + decoder_outputs[1:]  # Add hidden states and attention if they are here
+
+        if lm_labels is not None:
+            loss_fct = nn.CrossEntropyLoss()
+            masked_lm_loss = loss_fct(lm_logits.view(-1, self.config.vocab_size), lm_labels.view(-1))
+            decoder_outputs = (masked_lm_loss,) + decoder_outputs
+        return decoder_outputs + encoder_outputs
+
+    def output_layer(self, x):
+        return F.linear(x, self.model.shared.weight)
+
+
+class BartForSequenceClassification(PretrainedBartModel):
+    eos_token = 2
+
+    def __init__(self, config: BartConfig, **kwargs):
+        super().__init__(config, **kwargs)
+        self.model = BartModel(config)
+        self.classification_head = BARTClassificationHead(
+            config.d_model, config.d_model, config.num_labels, config.classif_dropout,
+        )
+        self.loss_fn = nn.CrossEntropyLoss()
+
+    def forward(self, input_ids, *args, **kwargs):
```

Comment:
Same as above. Also note a tricky thing: additional arguments (e.g. `labels`) should be added after the arguments of `self.model` otherwise torchscript can break on some edge cases

Reply:
do you mean that an `labels` (for example) should be passed to `self.model` even if `self.model` doesn't use it?